src/subspaceClient.ts

```diff
@@ -0,0 +1,114 @@
+import type { AddressOrPair } from "@polkadot/api/types";
+import type { Signer } from "@polkadot/types/types";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { hexToU8a, u8aToHex } from "@polkadot/util";
+import { Identity } from "./identity";
+import { appSettings } from "./config";
+
+/**
+ * @name SubspaceClient
+ * @summary  The SubspaceClient class expose methods to
+ * Put and Get an Object from Subspace network using the Storage API.
+ */
+export class SubspaceClient {
+  /**
+   * @name connect
+   * @summary Connect to Subspace network RPC nodes to put and get objects using the Subspace Storage API.
+   * @param identity A previously created Identity Object.
+   * @param wsRpcEndpoint Optional, by default it will connect to a public Subspace client rpc node.
+   * @param wsRpcEndpointFarmer Optional, by default it will connect to a public Subspace farmer rpc node.
+   * @return Promise<SubspaceClient>
+   */
+  public static async connect(
+    identity: Identity,
+    wsRpcEndpoint?: string,
+    wsRpcEndpointFarmer?: string
+  ): Promise<SubspaceClient> {
+    const farmerProvider = new WsProvider(
```

Comment:
I just noticed you use `WsProvider` form Polkadot.js for connection to farmer.
Honestly I'm surprised it works, but since it does, why not I guess :slightly_smiling_face:

Reply:
I was using another [lib](https://github.com/elpheria/rpc-websockets ) to connect that also worked. But after a while, I realized that `WsProvider` had this `send` method. 
It saves us a dependency